DancingGoat/Controllers/ProductController.cs

```diff
@@ -12,17 +13,17 @@ public class ProductController : ControllerBase
         [Route("{urlSlug}")]
         public async Task<ActionResult> Detail(string urlSlug)
         {
-            var response = await client.GetItemsAsync<object>(new EqualsFilter("elements.url_pattern", urlSlug), new InFilter("system.type", "brewer", "coffee"));
+            var item = (await client.GetItemsAsync<object>(new EqualsFilter("elements.url_pattern", urlSlug), new InFilter("system.type", "brewer", "coffee"))).Items.FirstOrDefault();
 
-            if (response.Items.Count == 0)
+            if (item == null)
             {
                 throw new HttpException(404, "Not found");
             }
             else
             {
                 ViewBag.FreeTasteRequested = TempData["formSubmited"] ?? false;
                 ViewBag.UrlSlug = urlSlug;
-                return View(response.Items[0]);
+                return View(item.GetType().Name, item);
```

Comment:
Because MVC has apparently no logic to determine the required view in such a situation. It only can pick the 'conventional view' which is ~/Views/[controller name]/[action method name].cshtml as a fallback when no explicit view is selected.